app/views/courses/TeacherClassesView.coffee

```diff
@@ -143,11 +144,18 @@ module.exports = class TeacherClassesView extends RootView
     latestHourTime = new Date() - -21 * 24 * 60 * 60 * 1000
     @upcomingOfficeHours = _.sortBy (oh for oh in officeHours when earliestHourTime < oh.time < latestHourTime), 'time'
     @howManyOfficeHours = if storage.load('hide-office-hours') then 'none' else 'some'
-    me.getClientCreatorPermissions()?.then(() => 
+    me.getClientCreatorPermissions()?.then(() =>
       @calculateQuestCompletion()
       @render?()
     )
 
+    administratingTeacherIds = me.get('administratingTeachers') || []
+
+    @administratingTeachers = new Users()
+    if administratingTeacherIds.length > 0
+      req = @administratingTeachers.fetchByIds(administratingTeacherIds)
+      @supermodel.trackRequest req
```

Comment:
We generally try to avoid using supermodel for any new changes and use Promises/async-await instead. Fine for now, but something to keep in mind.

Reply:
Thanks for the heads up - do you have a good example of what this should look like?  My understanding is that supermodel tracks requests to load collection / model data and that supermodel's events control the loading progress indicator, so we would have to use it to make sure the loading indicator is accurate.

Reply:
I generally use promises and then call `@render` once the promise is resolved. Not sure if we have a good example of its integration with loading progress indicator.

Reply:
Ok good to know thank you.  To summarize:

1. We're moving away from super model to using promises to track request state, calling rendering when promises resolve.
1. When promises are used alongside super model we do not integrate the two but rather call render after the promises resolve.
1. In these situations we do not show a loading indicator but rather render the page with the data after it loads.

Is this correct?

Reply:
I believe that's accurate. We currently also don't do any form of sophisticated approach to render once everything is loaded. I think we definitely have situations where different promises are resolving in non determinant orders and are calling render over and over. I believe this happens on the campaign view.

This the unlocked levels may appear. Then when user data arrives the locking algorithm runs and calls render again causing the levels to flash to locked.

Reply:
👍 👍